### PR TITLE
[docs] List components jsdocs

### LINKED ISSUE #1728: [doc]: JSDocs for Peregrine List components
**Describe the request**

Create or update comment blocks in for the Peregrine [List components][] so that they adhere to [JSDocs][] standards.

[jsdocs]: https://devdocs.io/jsdoc/
[list components]: https://github.com/magento/pwa-studio/tree/develop/packages/peregrine/lib/List

**Possible solutions**

Add JSDoc comment blocks to the Peregrine List components:

* List
* Items
* Item

See [useListState][] for a good example of JSDoc comments.

[useListState]: https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/List/useListState.js

### PR DESCRIPTION
**Pull request by dani97**
## Description

Added JsDocs For List Component

## Related Issue

Closes #1728 .

### Verification Stakeholders

@jcalcaben 

### Verification Steps

1. JSDocs should be generated for the component 

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/List/item.js [hunk 1]
```diff
@@ -47,6 +57,22 @@ const Item = props => {
...
 };
 
+/**
+ * props for {@link Item}
+ *
+ * @typedef props
+ *
+ * @property {Object} classes css classes prop for Item
+ * @property {string} classes.root css classes for Item root container
+ * @property {bool} hasFocus Is item has focus
```
Comment (jcalcaben) [inline packages/peregrine/lib/List/item.js]:
```suggestion
 * @property {bool} hasFocus Does the item have focus
```

File: packages/peregrine/lib/List/item.js [hunk 2]
```diff
@@ -47,6 +57,22 @@ const Item = props => {
...
+ * @property {bool} isSelected Is item selected
+ * @property {any} item item data
+ * @property {number} itemIndex index of item
+ * @property {func | string} render A render prop for the list item. A tagname string, such as `"div"`, is also valid
+ * @property {func} setFocus A callback for setting focus
+ * @property {number | string} uniqueId unique Id given forthe item
```
Comment (jcalcaben) [inline packages/peregrine/lib/List/item.js]:
```suggestion
 * @property {number | string} uniqueId unique Id given for the item
```

File: packages/peregrine/lib/List/items.js [hunk 3]
```diff
@@ -5,6 +5,16 @@ import iterable from '../validators/iterable';
 import Item from './item';
 import { useListState } from './useListState';
 
+/**
+ * The **Item** is a container holding all the items
```
Comment (jcalcaben) [inline packages/peregrine/lib/List/items.js]:
```suggestion
 * The **Items** component is a container holding all the items
```

**Comment by dani97** [comment]
@jcalcaben  made changes 👍

File: packages/peregrine/lib/List/item.js [hunk 4]
```diff
@@ -47,6 +57,22 @@ const Item = props => {
...
 
+/**
+ * props for {@link Item}
+ *
+ * @typedef props
+ *
+ * @property {Object} classes css classes prop for Item
+ * @property {string} classes.root css classes for Item root container
+ * @property {bool} hasFocus Does the item have focus
+ * @property {bool} isSelected Does the item is selected
```
Comment (jcalcaben) [inline packages/peregrine/lib/List/item.js]:
```suggestion
 * @property {bool} isSelected Is the item currently selected
```

Comment (dani97) [reply packages/peregrine/lib/List/item.js]: Updated 👍

**Pull request merged**